engine/access/rpc/backend/backend_transactions.go

```diff
@@ -66,6 +66,7 @@ func (b *backendTransactions) SendTransaction(
 	// store the transaction locally
 	err = b.transactions.Store(tx)
 	if err != nil {
+		// TODO: why would this be InvalidArgument?
 		return status.Error(codes.InvalidArgument, fmt.Sprintf("failed to store transaction: %v", err))
```

Comment:
Any idea why this is `InvalidArgument` instead of `Internal`?

Reply:
I thought this was to differentiate it on the client side? let me double check

Reply:
It seems odd to return `InvalidArguement` if the tx is actually valid and the error was writing it to the db.

Thinking about this some more, I don't think we should return an error at all. As is the client may think the TX failed when it really succeeded, which could result in a double payment.